src/dune_rules/context.ml

```diff
@@ -569,7 +555,7 @@ module rec Instantiate : sig
   val instantiate : Context_name.t -> instance Memo.t
 end = struct
   let instantiate_impl name : instance Memo.t =
-    let env = Global.env () in
+    let env = Dune_cli.Global.env () in
```

Comment:
How to lift this?

I generalized this whole module to use a pair of an env and a context name but the changes propagate quite far and I am unsure this is the api we want in the end.